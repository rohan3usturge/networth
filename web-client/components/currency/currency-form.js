import { withTranslation } from "react-i18next";
import Select from "react-select";
import { CURRENCY_UTILS } from "./../utils";

const Wrapper = ({ currencies, onChange, selectedCurrency, t }) => {
  const handleChange = ({ value }) => {
    if (onChange) {
      onChange(value);
    }
  };

  const getClass = (curr) => {
    const props = CURRENCY_UTILS.currencyCodeToFaClass[curr];
    return props ? props.class : "";
  };

  const getLabel = (curr) => {
    const props = CURRENCY_UTILS.currencyCodeToFaClass[curr];
    return props ? props.label : "";
  };

  const options = currencies.map((curr) => ({
    value: curr,
    label: (
      <span id={`option-${curr}`}>
        <i className={`mr-2 fa ${getClass(curr)}`}></i>
        {curr} - {getLabel(curr)}
      </span>
    ),
  }));

  const selectedValue = options.find((o) => o.value === selectedCurrency);

  return (
    <div className="card mb-3">
      <div className="card-body">
        <div className="form-group">
          <label className="h6 mb-2" htmlFor="currency-select">
            {t("currency")}
          </label>
          <Select
            id="currency-seletor"
            defaultValue={selectedValue}
            onChange={handleChange}
            options={options}
          />
        </div>
      </div>
    </div>
  );
};

const CurrencyForm = withTranslation()(Wrapper);

export { CurrencyForm };
