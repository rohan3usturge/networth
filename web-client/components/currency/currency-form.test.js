import { shallow } from "enzyme";
import Select from "react-select";
import { Wrapper } from "./currency-form";

/** @test {Title Component} */
describe("CurrencyForm", () => {
  it("should render properly and match snapshot", () => {
    const translations = {
      currency: "Currency",
    };

    const currencies = ["CAD", "AUS"];
    const wrapper = shallow(
      <Wrapper
        selectedCurrency="CAD"
        currencies={currencies}
        t={(currency) => translations[currency]}
      />
    );
    const label = wrapper.find("label");
    const selectProps = wrapper.find(Select).props();
    expect(selectProps.defaultValue.value).toBe("CAD");
    expect(selectProps.options.length).toBe(currencies.length);
    expect(label).toHaveLength(1);
    expect(label.text()).toBe(translations.currency);
    expect(wrapper).toMatchSnapshot();
  });
});
