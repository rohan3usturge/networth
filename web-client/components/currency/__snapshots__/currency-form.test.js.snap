// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyForm should render properly and match snapshot 1`] = `
<div
  className="card mb-3"
>
  <div
    className="card-body"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 mb-2"
        htmlFor="currency-select"
      >
        Currency
      </label>
      <StateManager
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={
          Object {
            "label": <span
              id="option-CAD"
            >
              <i
                className="mr-2 fa fa-dollar-sign"
              />
              CAD
               - 
              Canadian Dollar
            </span>,
            "value": "CAD",
          }
        }
        id="currency-seletor"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": <span
                id="option-CAD"
              >
                <i
                  className="mr-2 fa fa-dollar-sign"
                />
                CAD
                 - 
                Canadian Dollar
              </span>,
              "value": "CAD",
            },
            Object {
              "label": <span
                id="option-AUS"
              >
                <i
                  className="mr-2 fa "
                />
                AUS
                 - 
                
              </span>,
              "value": "AUS",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
